fix: Documentation workflow - shell syntax error in source docs generation

## Workflow Failure Report

**Workflow:** Documentation #427
**Run URL:** https://github.com/robinmordasiewicz/f5xc-xcsh/actions/runs/20670200580
**Triggered by:** Release workflow completion
**Failed Step:** Generate source docs with real output

## Error Details
```
/Users/runner/work/_temp/605ed038-c94c-466b-9c07-81458af8f528.sh: line 30: syntax error near unexpected token `('
```

## Root Cause Analysis
The error occurs when passing complex multi-line output (containing special characters like parentheses) to the Python script `scripts/generate-source-docs.py` via shell arguments. The build output includes error messages with parentheses that break shell parsing:
```
Error: Cannot find module '/private/var/folders/bp/kmfmhnl95kx1c8x321z7twbw0000gn/T/tmp.LfVBo1atJl/xcsh/dist/index.js'
    at Function._resolveFilename (node:internal/modules/cjs/loader:1383:15)
```

## Additional Context
- The previous step "Test source build and capture output" captured a failed build due to missing specs
- These error messages containing special characters (`(`, `)`) were passed to the next step
- The shell interpreted the parentheses as subshell syntax, causing the syntax error

## Suggested Fix
1. Escape special characters in the captured output before passing to shell
2. Or use a different method to pass multi-line output (temp files, environment variables, base64 encoding)
3. Or handle build failures gracefully before attempting to generate docs

## Note
This is a separate issue from the asset naming fix in PR #459 which resolved the Windows asset download failure.

---
🤖 Auto-generated by /ship workflow

--- a/.github/workflows/docs.yml
+++ b/.github/workflows/docs.yml
@@ -679,6 +679,7 @@ jobs:
           echo "prereq_output<<EOFPREREQ" >> $GITHUB_OUTPUT
           echo "$PREREQ_OUTPUT" >> $GITHUB_OUTPUT
           echo "EOFPREREQ" >> $GITHUB_OUTPUT
+          echo "$PREREQ_OUTPUT" > $GITHUB_WORKSPACE/prereq-output.txt
 
           # Extract Node version for template
           NODE_VERSION_SHORT=$(echo "$NODE_VER" | sed 's/^v//')
@@ -701,6 +702,7 @@ jobs:
           echo "clone_output<<EOFCLONE" >> $GITHUB_OUTPUT
           echo "$CLONE_OUTPUT" >> $GITHUB_OUTPUT
           echo "EOFCLONE" >> $GITHUB_OUTPUT
+          echo "$CLONE_OUTPUT" > $GITHUB_WORKSPACE/clone-output.txt
 
           cd xcsh
 
@@ -715,6 +717,7 @@ jobs:
           echo "install_output<<EOFINSTALL" >> $GITHUB_OUTPUT
           echo "$INSTALL_OUTPUT" >> $GITHUB_OUTPUT
           echo "EOFINSTALL" >> $GITHUB_OUTPUT
+          echo "$INSTALL_OUTPUT" > $GITHUB_WORKSPACE/install-output-source.txt
 
           # Capture build output
           echo "Building xcsh..."
@@ -730,6 +733,7 @@ jobs:
           echo "build_output<<EOFBUILD" >> $GITHUB_OUTPUT
           echo "$BUILD_OUTPUT" >> $GITHUB_OUTPUT
           echo "EOFBUILD" >> $GITHUB_OUTPUT
+          echo "$BUILD_OUTPUT" > $GITHUB_WORKSPACE/build-output.txt
 
           # Capture version output
           echo "Verifying build..."
@@ -742,6 +746,7 @@ jobs:
           echo "version_output<<EOFVERSION" >> $GITHUB_OUTPUT
           echo "$VERSION_OUTPUT" >> $GITHUB_OUTPUT
           echo "EOFVERSION" >> $GITHUB_OUTPUT
+          echo "$VERSION_OUTPUT" > $GITHUB_WORKSPACE/version-output.txt
 
           # Cleanup
           cd /
@@ -754,11 +759,11 @@ jobs:
         run: |
           python scripts/generate-source-docs.py \
             --node-version "${{ steps.source-build.outputs.node_version }}" \
-            --prereq-output "${{ steps.source-build.outputs.prereq_output }}" \
-            --clone-output "${{ steps.source-build.outputs.clone_output }}" \
-            --install-output "${{ steps.source-build.outputs.install_output }}" \
-            --build-output "${{ steps.source-build.outputs.build_output }}" \
-            --version-output "${{ steps.source-build.outputs.version_output }}" \
+            --prereq-output "$(cat prereq-output.txt)" \
+            --clone-output "$(cat clone-output.txt)" \
+            --install-output "$(cat install-output-source.txt)" \
+            --build-output "$(cat build-output.txt)" \
+            --version-output "$(cat version-output.txt)" \
             --output docs/install/source.md
 
       - name: Generate command documentation

